readthedocs/projects/views/private.py

```diff
@@ -154,6 +156,15 @@ def project_version_detail(request, project_slug, version_slug):
     if request.method == 'POST' and form.is_valid():
         version = form.save()
         if form.has_changed():
+            if 'slug' in form.changed_data and version.slug not in (STABLE, LATEST):
+                # Rebuild symlinks to the new slug, remove the old slug's symlink
+                # Latest and Stable would appear here because they are disabled on the form
+                log.info('Rebuilding symlinks for %s. A version slug changed', version.project)
+                broadcast(type='app', task=tasks.symlink_project, args=[version.project.pk])
```

Comment:
We are already calling this on `Version.save` which is called from line 157 (`form.save()`).

I'm not 100% sure, but I think that if the version is set as `active` a build is triggered for that version when it's saved also.

In case of `activate` the version _and_ changing the `slug` we will be doing this twice. _I_, _think_ :)

Reply:
> We are already calling this on Version.save which is called from line 157 

Good catch

> I'm not 100% sure, but I think that if the version is set as active a build is triggered for that version when it's saved also.

I don't see this anywhere and in my testing this did not occur.